tests/queries/tests.py

```diff
@@ -3800,3 +3800,77 @@ def test_ticket_23622(self):
             Ticket23605A.objects.filter(qx),
             [a2], lambda x: x
         )
+
+
+class ExcludeCombineTests(TestCase):
```

Comment:
Would it make sense to move this to a new `queries/test_xxx.py` file? I think we need to break up the file and maybe an incremental approach when adding new tests would work best?